src/modules/avformat/producer_avformat.c

```diff
@@ -1332,20 +1429,48 @@ static int convert_image( producer_avformat self, AVFrame *frame, uint8_t *buffe
 	}
 	else
 	{
+		int i, c;
+		AVPicture output;
+		struct sliced_pix_fmt_conv_t ctx =
+		{
+			.flags = flags,
+			.width = width,
+			.height = height,
+			.frame = frame,
+			.output = &output,
+			.dst_format = AV_PIX_FMT_YUYV422,
+			.src_colorspace = self->yuv_colorspace,
+			.dst_colorspace = profile->colorspace,
+			.src_full_range = self->full_luma,
+			.dst_full_range = 0,
+		};
 #if defined(FFUDIV) && (LIBAVFORMAT_VERSION_INT >= ((55<<16)+(48<<8)+100))
-		struct SwsContext *context = sws_getContext( width, height, src_pix_fmt,
-			width, height, AV_PIX_FMT_YUYV422, flags | SWS_FULL_CHR_H_INP, NULL, NULL, NULL);
+		ctx.src_format = src_pix_fmt;
 #else
-		struct SwsContext *context = sws_getContext( width, height, pix_fmt,
-			width, height, AV_PIX_FMT_YUYV422, flags | SWS_FULL_CHR_H_INP, NULL, NULL, NULL);
+		ctx.src_format = pix_fmt;
 #endif
-		AVPicture output;
-		avpicture_fill( &output, buffer, AV_PIX_FMT_YUYV422, width, height );
-		if ( !set_luma_transfer( context, self->yuv_colorspace, profile->colorspace, self->full_luma, 0 ) )
-			result = profile->colorspace;
-		sws_scale( context, (const uint8_t* const*) frame->data, frame->linesize, 0, height,
-			output.data, output.linesize);
-		sws_freeContext( context );
+		ctx.src_desc = av_pix_fmt_desc_get( ctx.src_format );
+		ctx.dst_desc = av_pix_fmt_desc_get( ctx.dst_format );
+
+		avpicture_fill( ctx.output, buffer, ctx.dst_format, width, height );
+
+		if ( getenv("MLT_AVFORMAT_SLICED_PIXFMT") )
```

Comment:
>> i though about it, but it will require some code to check if sws init condition changes

> I see. So maybe it could be a future optimization

 did some checks (https://github.com/max-verem/mlt/commit/30b52552da686ca3884b2f600e40e09147497e2f):
```
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=407 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=416 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=394 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=413 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=1002 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=375 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=359 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=374 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=358 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=353 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=355 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=356 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=353 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=391 us
Current Position:         15
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=373 us
sliced_h_pix_fmt_conv_proc:1318: DUR(sws_context_init)=1582 us
convert_image:1485: DUR(convert_image)=10054 us
producer_get_image:1984: DUR(producer_get_image)=15155 us
```
average 356 us